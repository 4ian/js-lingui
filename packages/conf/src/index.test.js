import path from "path"
import { validate } from "jest-validate"
import {
  getConfig,
  replaceRootDir,
  catalogMigration,
  configValidation
} from "@lingui/conf"
import { mockConsole, getConsoleMockCalls } from "@lingui/jest-mocks"

describe("@lingui/conf", function() {
  it("should return default config", function() {
    expect.assertions(2)

    mockConsole(console => {
      const config = getConfig({
        cwd: path.resolve(__dirname, path.join("fixtures", "valid"))
      })
      expect(console.warn).not.toBeCalled()
      expect(config).toMatchSnapshot()
    })
  })

  it("should validate `locale`", function() {
    expect.assertions(2)

    mockConsole(console => {
      getConfig()
      expect(console.warn).not.toBeCalled()
      expect(console.error).toBeCalledWith(
        expect.stringContaining("No locales defined")
      )
    })
  })

  it("should replace <rootDir>", function() {
    const config = replaceRootDir(
      {
        boolean: false,
        catalogs: [
          {
            path: "<rootDir>/locales/{locale}/messages",
            include: ["<rootDir>/src"],
            exclude: ["<rootDir>/ignored"]
          }
        ]
      },
      "/Root"
    )

    expect(config.boolean).toEqual(false)
    expect(config.catalogs).toEqual([
      {
        path: "/Root/locales/{locale}/messages",
        include: ["/Root/src"],
        exclude: ["/Root/ignored"]
      }
    ])
  })

  describe("catalogMigration", function() {
    it("should show deprecation warning", function() {
      mockConsole(console => {
        validate(
          {
            localeDir: "./locale",
            srcPathDirs: ["./src"],
            srcPathIgnorePatterns: ["/node_modules/"]
          },
          configValidation
        )
        expect(getConsoleMockCalls(console.warn)).toMatchSnapshot()
      })
    })

    it("shouldn't provide default config if no obsolete config is defined", function() {
      const config = {}

      expect(catalogMigration(config)).toEqual({})
    })

    it("should normalize string localeDir", function() {
      const config = {
        localeDir: "./locales"
      }

      expect(catalogMigration(config)).toEqual({
        catalogs: [
          {
            path: "locales/{locale}/messages",
            include: ["<rootDir>"],
            exclude: ["*/node_modules/*"]
          }
        ]
      })
    })

    it("should migrate srcPathDirs and srcPathIgnorePatterns", function() {
      const config = {
        localeDir: "locales",
        srcPathDirs: ["src"],
        srcPathIgnorePatterns: ["src/node_modules/"]
      }

      expect(catalogMigration(config)).toEqual({
        catalogs: [
          {
            path: "locales/{locale}/messages",
            include: ["src"],
            exclude: ["src/node_modules/"]
          }
        ]
      })
    })
  })
})
