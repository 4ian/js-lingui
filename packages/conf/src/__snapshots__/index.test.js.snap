// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lingui/conf catalogMigration should show deprecation warning 1`] = `
● Deprecation Warning:

 Option localeDir is deprecated. Configure source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": [
  {
    "path": "locale/{locale}/messages",
    "include": [
      "./src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathDirs is deprecated. Configure source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": [
  {
    "path": "locale/{locale}/messages",
    "include": [
      "./src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathIgnorePatterns is deprecated. Configure excluded source paths using catalogs instead.
    
    @lingui/cli now treats your current configuration as:
    
    {
      "catalogs": [
  {
    "path": "locale/{locale}/messages",
    "include": [
      "./src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }
    
    Please update your configuration.
    

Documentation: https://lingui.js.org/ref/conf.html
`;

exports[`@lingui/conf should return default config 1`] = `
Object {
  catalogs: Array [
    Object {
      exclude: Array [
        */node_modules/*,
      ],
      include: Array [
        /Users/tricoder/Projects/lingui/js-lingui/packages/conf/src/fixtures/valid,
      ],
      path: /Users/tricoder/Projects/lingui/js-lingui/packages/conf/src/fixtures/valid/locale/{locale}/messages,
    },
  ],
  compileNamespace: cjs,
  extractBabelOptions: Object {
    plugins: Array [],
    presets: Array [],
  },
  fallbackLocale: ,
  format: po,
  locales: Array [
    en,
    cs,
  ],
  pseudoLocale: ,
  rootDir: /Users/tricoder/Projects/lingui/js-lingui/packages/conf/src/fixtures/valid,
  sourceLocale: ,
}
`;
